scripts/mbedtls_dev/bignum_mod_raw.py

```diff
@@ -40,6 +43,385 @@ class BignumModRawTarget(test_data_generation.BaseTarget, metaclass=ABCMeta):
+                        "b17bf2884d65d7e4f927db2a00ad7f6a2573d6503b5fd9afa43a36435db230e5f0d840"
+                        "861"
+                    ),
+                    (
+                        "53ddb006b78da1601483110f11717b914c8cb6190c4314766c029518105d8d5eb57171"
+                        "e63"
+                    ),
+                ),
+                (
+                    (
+                        "c0959708ad338b97881c844a715c577fbc767f9c134c768cb1d8ec450cd8fcc8aaef42"
+                        "99e"
+                    ),
+                    (
+                        "d48e00413812626969d8e28a1067d1c217385f7d1b4f646557a5024b04554533b2e17b"
+                        "e3d"
+                    ),
+                ),
+            ],
+        ),
+        (
+            (
+                "2A94608DE88B6D5E9F8920F5ABB06B24CC35AE1FBACC87D075C621C3E2833EC902713E40"
+                "F51E3B3C214EDFABC451"
+            ),
+            (
+                "0x2a94608de88b6d5e9f8920f5abb06b24cc35ae1fbacc87d075c621c3e2833ec902713e40"
+                "f51e3b3c214edfabc451 is (dec) 99999999977^10"
+            ),
+            [
+                (
+                    (
+                        "1ddec6e06c2028fca75c93545380715797b1a0ef0d62a1d46e41da6bb7318f4c7bff84"
+                        "3397453fa8c58e1db479d2"
+                    ),
+                    (
+                        "1c96ebcfa1dd0ef3cc2c18dd9d25a1c7d05ac9957abfc82623dc09f56608efaa5f4631"
+                        "64a99c33385bd51f37b7ff"
+                    ),
+                ),
+                (
+                    (
+                        "1c69cf95d47246c5cb87185bfd347b9f13e71f4513055dbbd488c0da31334a0e1242db"
+                        "19d4607e56a12b6f7ed6e9"
+                    ),
+                    (
+                        "1ff83a67f26bacb0d2b4632704410890719895b34fc073b5c88120b1eebe7947a82190"
+                        "7446b91a318732f38c2560"
+                    ),
+                ),
+                (
+                    (
+                        "1fb94a8e70af7a8c4bd7158127015c49792524121bfd70115952ce0e88d358f7e178f7"
+                        "869d761cfb7ce9ac05ed24"
+                    ),
+                    (
+                        "26183f151f6dfb6cbc0f604fc86da6f236e214b9523433fcdc9ce4e9f1d9ae160f394e"
+                        "f2a2e0407f3c7c9ce598b7"
+                    ),
+                ),
+                (
+                    (
+                        "133a5029cceb75614fa1a1da5dc2b6ffd034c227a3923c9338cedc28caf194c32c95f1"
+                        "52379e725a8bff6fdd7c99"
+                    ),
+                    (
+                        "129511d6de42ab2572916fff044f78abcce684996637baeea6100aedae21457484c9b2"
+                        "62dbaf0693de626cacc629"
+                    ),
+                ),
+            ],
+        ),
+    ]
+
+    def __init__(self, val_a: str, val_b: str, modulus: str, mod_desc: str) -> None:
```

Comment:
Modying the init parameters, comes with an inherent cost of imposing those argument and this design to other tests which will need to inherit from `BignumModRawOperation` .

I would keep the `BignumModRawOperation` minimal and decide on a way of handling extra parameters. This can be achieved by many ways in python such as inheritance, decorators, or simply an overridable boilerplate method with kwargs.

eg

~~~~~~
Class SomeClas:

    def __init__(self, val_a: str, val_b: str) -> None:
        super.init(val_a, val_b)
        self.handle_extra_args()

    def handle_extra_args(**kwargs):
        for key, value in kwargs.items():
             setattr(self, 'key', value)
~~~~~~

Then if a subclass needs to change the members, could just override `handle_extra_args()` without rewriting the init.